_examples/blockchain.adoc

```diff
@@ -0,0 +1,107 @@
+= Blockchain Analysis Notebook
+:page-link: blockchain
+:page-weight: 101
+:page-labels: [Python, Jupyter, spark-notebook]
+:page-layout: application
+:page-menu_template: menu_tutorial_application.html
+:page-description: The blockchain notebooks are examples of how to explore graph data using GraphX and GraphFrames on top of OpenShift using Spark. It uses the real Bitcoin blockchain data to create transaction graph for the analysis.
+:page-project_links: ["https://github.com/Jiri-Kremser/bitcoin-insights"]
+
+[[introduction]]
+== Introduction
+
+Cryptocurrencies attract various groups of people. Among other it could be
+investors, people from retail, tech enthusiasts, crypto-anarchists, etc.
+We are not going to focus on anything else than the raw technology behind the
+Blockchain, leaving aside all the ideology and hype that comes with the Bitcoin.
+
+The https://spark.apache.org/graphx/[GraphX] based notebook uses the https://github.com/spark-notebook/spark-notebook[spark-notebook] as the notebook technology,
+while the second one uses https://graphframes.github.io/[GraphFrames] and Jupyter notebook. Also the second
+notebook connects to an existing spark cluster that was created by Oshinko tools.
+
+[[architecture]]
+== Architecture
+
+A Jupyter/Spark notebook is an interactive compute environment that interleaves
+documentation, code examples, and graphical output.
+
+Similarly as our other notebooks the notebooks are the drivers that interactively
+sends the commands to Spark cluster for the evaluation. They can either spawn
+its own standalone Spark or connect to an external one.
+
+[[installation]]
+== Install and run the notebook
+
+Installing our notebook is very simple:  we just need to create an OpenShift project and install and run the notebook application.
+Make sure you're logged in to OpenShift and have selected the right project, and then execute the following two commands:
+
+....
+oc new-app jkremser/bitcoin-notebook:tutorial-1.0.0
+oc expose svc/bitcoin-notebook
+....
+
+for the Jupyter notebook and
+
+....
+oc new-app jkremser/bitcoin-spark-notebook:tutorial-1.0.0
+oc expose svc/bitcoin-spark-notebook
+....
+
+Once both routes are exposed the both notebooks should be available. You
+can find the URLs if you run `oc get routes`.
+It is also possible for the Jupyter notebook to set
+environmental variable that assures it will connect to an external Spark cluster.
+
+To do that you can optionally:
+`oc new-app jkremser/bitcoin-notebook:tutorial-1.0.0 -e SPARK_MASTER=spark://c:7077`
+
+Where `c` is the hostname of the Spark cluster.
+
+[[usage]]
+== Usage
+Both notebooks comes with some prepared data in the parquet files, but part of the
+project is also the https://github.com/Jiri-Kremser/bitcoin-insights/tree/master/parquet-converter[parquet-converter]
+that takes the raw Blockchain data and converts
+it to the format that is digestible by Spark. It assumes the official bitcoin wallet to be installed
```

Comment:
i'm not quite following the sentence starting with "It assumes", what did you mean to say here?

Reply:
it = the converter

the converter takes the files downloaded by the default Bitcoin client as its input and create parquet files with nodes and edges as the output. I wanted to mention that it's required for the converter that the official bitcoin client is installed (either `bin/bitcoin-qt` or `bin/bitcoind`). 

However, the docker images contain some pre-baked data so if user is not interested in some particular time period, it should just work. Otherwise, they should use the converter to prepare own data for the analysis. The makefile expects some env variables with the paths to the raw blk files (https://git.io/vFg01).

Would it be better if I put it this way:
"
The converter assumes the official bitcoin client to be installed, because it downloads the whole Blockchain (~150 GB) into `~/.bitcoin/blocks/`. Those files located in this directory are then the input to the converter, for more details see the [Makefile](https://git.io/vFg01).
"
?

Reply:
yeah, i like the way you put it. i think it would also be worthwhile to explain a little more. 

this is good info imo
> However, the docker images contain some pre-baked data so if user is not interested in some particular time period, it should just work. Otherwise, they should use the converter to prepare own data for the analysis. The makefile expects some env variables with the paths to the raw blk files (https://git.io/vFg01).